java/statestore/src/test/java/sleeper/statestore/transactionlog/TransactionLogSnapshotTestBase.java

```diff
@@ -0,0 +1,239 @@
+        tableProperties.set(TABLE_ID, tableId);
+        tableProperties.set(TABLE_NAME, tableName);
+        tableProperties.set(STATESTORE_CLASSNAME, DynamoDBTransactionLogStateStore.class.getName());
+        fileTransactionStoreByTableId.put(tableId, new InMemoryTransactionLogStore());
+        partitionTransactionStoreByTableId.put(tableId, new InMemoryTransactionLogStore());
+        return tableProperties;
+    }
+
+    protected TransactionLogSnapshotMetadata filesSnapshot(TableProperties table, long transactionNumber) {
+        return TransactionLogSnapshotMetadata.forFiles(getBasePath(instanceProperties, table), transactionNumber);
+    }
+
+    protected TransactionLogSnapshotMetadata partitionsSnapshot(TableProperties table, long transactionNumber) {
+        return TransactionLogSnapshotMetadata.forPartitions(getBasePath(instanceProperties, table), transactionNumber);
+    }
+
+    protected String filesSnapshotPath(TableProperties table, long transactionNumber) {
+        return filesSnapshot(table, transactionNumber).getPath();
+    }
+
+    protected String partitionsSnapshotPath(TableProperties table, long transactionNumber) {
+        return partitionsSnapshot(table, transactionNumber).getPath();
+    }
+
+    protected boolean filesSnapshotFileExists(TableProperties table, long transactionNumber) throws IOException {
+        return fs.exists(new Path(filesSnapshot(table, transactionNumber).getPath()));
+    }
+
+    protected boolean partitionsSnapshotFileExists(TableProperties table, long transactionNumber) throws IOException {
+        return fs.exists(new Path(partitionsSnapshot(table, transactionNumber).getPath()));
+    }
+
+    protected void deleteFilesSnapshotFile(TableProperties table, long transactionNumber) throws Exception {
+        fs.delete(new Path(filesSnapshot(table, transactionNumber).getPath()), false);
+    }
+
+    protected void deletePartitionsSnapshotFile(TableProperties table, long transactionNumber) throws Exception {
+        fs.delete(new org.apache.hadoop.fs.Path(partitionsSnapshot(table, transactionNumber).getPath()), false);
+    }
+
+    protected Stream<String> tableFiles(TableProperties tableProperties) throws Exception {
+        Path tableFilesPath = new Path(getBasePath(instanceProperties, tableProperties));
+        if (!fs.exists(tableFilesPath)) {
+            return Stream.empty();
+        }
+        RemoteIterator<LocatedFileStatus> iterator = fs.listFiles(tableFilesPath, true);
+        List<String> files = new ArrayList<>();
+        while (iterator.hasNext()) {
+            files.add(iterator.next().getPath().toUri().toString());
+        }
+        return files.stream();
+    }
+
+    protected static String getBasePath(InstanceProperties instanceProperties, TableProperties tableProperties) {
+        return instanceProperties.get(FILE_SYSTEM)
+                + instanceProperties.get(DATA_BUCKET) + "/"
+                + tableProperties.get(TableProperty.TABLE_ID);
```

Comment:
This is duplicating DynamoDBTransactionLogSnapshotStore.getBasePath. It doesn't look like that's actually tested anywhere. We could call that directly instead of using this, but add a test for that method as well?